### DISCUSSION TITLE
Algolia Event Tracking

### DISCUSSION THREAD

**Discussion by JSShant** [algolia/instantsearch#5816] [Q&A]
Hey everyone using Algolia Insights Middleware to do some tracking for the hits from Algolia however it seems that the view events from just loading onto the page are duplicatied, if i load into a page with 1 hit, im seeing 4 view events in the algolia dashboard, is there a way to work around this or is this intended, thankyou!

**Comment by Haroenv** [comment]
Are you seeing multiple view events being sent from your application as well? Are you using useHits or useInfiniteHits multiple times in the app too, or only once?

**Reply by JSShant** [reply]
yup on the onEvent when i console log the event im getting that numerous times, Im using one instance of useInfinitehits and passing hits into different components that require it.

**Reply by Haroenv** [reply]
do you have a simple reproduction somewhere? this should normally not send view multiple times, but could be a bug

**Reply by JSShant** [reply]
Will attempt to produce a reproduction, however im thinking if it has something to do with the send event being part of the render function for inifinitehits : https://github.com/algolia/instantsearch/blob/master/packages/instantsearch.js/src/connectors/infinite-hits/connectInfiniteHits.ts

**Reply by Haroenv** [reply]
Yes, this. is likely the case if there's any duplication happening, or part of sendEventForHits directly, where deduplication happens now. However there's definitely no deduplication happening across widget boundaries (today)

**Comment by deadjdona** [comment]
but today...
I see same events doubled in logs with few ms difference, when sending single view event from chrome console with window.aa
first it pings algolia.io url, then xhr it, and in api logs both events exists
